### ISSUE TITLE
Frozen ResNet Blocks and BatchNorms

### ISSUE THREAD

**Issue by andy-96** [aliyun/conditional-lane-detection#36]
Hi all,

does somebody know why the first ResNet Block and almost all BatchNorms are frozen? I have trained the model once without freeze and the performance was a bit worse, so what is the intuition on the frozen layers?

[mmdet/models/backbones/resnet.py : 546 and 606
](https://github.com/aliyun/conditional-lane-detection/blob/master/mmdet/models/backbones/resnet.py)
```python
    def _freeze_stages(self):
        if self.frozen_stages >= 0:
            if self.deep_stem:
                self.stem.eval()
                for param in self.stem.parameters():
                    param.requires_grad = False
            else:
                self.norm1.eval()
                for m in [self.conv1, self.norm1]:
                    for param in m.parameters():
                        param.requires_grad = False

        for i in range(1, self.frozen_stages + 1):
            m = getattr(self, f'layer{i}')
            m.eval()
            for param in m.parameters():
                param.requires_grad = False

    def train(self, mode=True):
        super(ResNet, self).train(mode)
        self._freeze_stages()
        if mode and self.norm_eval:
            for m in self.modules():
                # trick: eval have effect on BatchNorm only
                if isinstance(m, _BatchNorm):
                    m.eval()
```
